src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -434,5 +435,18 @@ internal static void SetCurrentUICulture(CultureInfo cultureInfo)
                 Thread.CurrentUICulture.SetValue(thread, cultureInfo);
             }
         }
+
+        internal static class Assembly
+        {
+            private const string TypeName = "System.Reflection.Assembly";
+
+            internal static readonly Type Type = ReflectionUtil.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            internal static readonly MethodInfo GetTypeThrowOnErrorIgnoreCase = Type
```

Comment:
For methods with overloads we try and spell out the name here vs. using a different name.  For example; GetType_string_bool_bool